src/transport/msgs.rs

```diff
@@ -0,0 +1,65 @@
+use crate::transport::jsonrpc;
+use lightning::impl_writeable_msg;
+use lightning::ln::wire;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct RawLSPSMessage {
+	pub payload: String,
+}
+
+impl_writeable_msg!(RawLSPSMessage, { payload }, {});
+
+impl wire::Type for RawLSPSMessage {
+	fn type_id(&self) -> u16 {
+		37913
+	}
+}
+
+/// An [`lsps0_message`] message to be sent to or received from a peer.
+///
+/// [`lsps0_message`]: https://github.com/BitcoinAndLightningLayerSpecs/lsp/blob/49544d1a420cf32ef4726c526a29aa46a09202ff/LSPS0/README.md
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum LSPSMessage {
```

Comment:
Ah, I hadn't thought of doing that.  It probably makes sense, since most users of this library will be 'clients' who are making requests to the LSP.  I'm a bit biased because our usage of this library is intended to be from the LSP side.

I definitely want to support both use-cases with this library.

I'll keep that in mind and make the split if it seems like it would reduce complexity for the end-users.